compiler/src/test/resources/bugs/apps-1191_scatter_optional_input.json

```diff
@@ -0,0 +1,141 @@
+{
+  "class": "Workflow",
+  "cwlVersion": "v1.2",
+  "id": "apps-1191_scatter_optional",
+  "inputs": [
+    {
+      "default": [
+        1,
+        2
+      ],
+      "id": "scatter_input",
+      "type": {
+        "items": "int",
+        "type": "array"
+      }
+    },
+    {
+      "default": [
+        false,
+        true,
+        false,
+        true
+      ],
+      "id": "condition_arg",
+      "type": {
+        "items": "boolean",
+        "type": "array"
+      }
+    }
+  ],
+  "outputs": [
+    {
+      "id": "scatter_output",
+      "outputSource": "step1/scatter_output",
+      "pickValue": "all_non_null",
+      "type": {
+          "items": {
+            "items": [
+              "null",
+              "string"
+            ],
+            "type": "array"
+        },
+        "type": "array"
+      }
+    }
+  ],
+  "requirements": [
+    {
+      "class": "ScatterFeatureRequirement"
+    },
+    {
+      "class": "SubworkflowFeatureRequirement"
+    },
+    {
+      "class": "InlineJavascriptRequirement"
+    }
+  ],
+  "steps": [
+    {
+      "id": "step1",
+      "in": [
+        {
+          "id": "scatter_input",
+          "source": "scatter_input"
+        },
+        {
+          "id": "condition_input",
+          "source": "condition_arg"
+        }
+      ],
+      "out": [
+        "scatter_output"
+      ],
+      "run": {
+        "baseCommand": [
+          "echo",
+          ""
+        ],
+        "class": "CommandLineTool",
+        "id": "echo",
+        "inputs": [
+          {
+            "type": [
+              "null",
+              "string"
+            ],
+            "id": "undeclared_no_default",
+            "inputBinding": {
+              "position": 1,
+              "prefix": "\nthis should not appear: "
+            }
+          },
+          {
+            "type": [
+              "null",
+              "string"
+            ],
+            "id": "undeclared_with_default",
+            "inputBinding": {
+              "position": 2,
+              "prefix": "\nthis should appear:"
+            },
+            "default": "there is a default."
```

Comment:
These two task inputs are not declared in wf. However, the one with default will be added to the output and the one without default in position 1 will be skipped.

So the final scatter output is:
```
[[null," scatter number: 1 this should appear: there is a default.\n",null," scatter number: 1 this should appear: there is a default.\n"],[null," scatter number: 2 this should appear: there is a default.\n",null," scatter number: 2 this should appear: there is a default.\n"]]
```
https://staging.dnanexus.com/projects/Fy9QqgQ0yzZbg9KXKP4Jz6Yq/monitor/analysis/G9kX7GQ0yzZvKg77BP1YxxXv